_posts/2023-02-01-dask-on-flyte.md

```diff
@@ -0,0 +1,151 @@
+- The Python environment within the docker image associated task needs to have `flytekitplugins-dask` installed
+
+## How do things work under the hood?
+
+_Note: The following should only be informative and should not be required for pure users of the plugin, it might be helpful for easier debugging though_
+
+On a birdseye view, the following steps happen when a `dask` task is started from within Flyte:
+
+1. A `FlyteWorkflow` [Custom Resource (CR)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) is created in Kubernetes
+1. Flyte propeller (a [Kubernetes Operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)) picks up on the workflow creation
+1. The operator checks the task's spec, sees that the task is of type `dask`. It checks whether it has a plugin associated with it and will find the installed `dask` plugin
+1. The `dask` plugin (within Flyte propeller) picks up on the task defintion and creates a `DaskJob` Custom Resource ([docs](https://kubernetes.dask.org/en/latest/operator_resources.html#daskjob)) using the [dask-k8s-operator-go-client](https://github.com/bstadlbauer/dask-k8s-operator-go-client/)
+1. The [dask operator](https://kubernetes.dask.org/en/latest/operator.html) picks up on the `DaskJob` resource and run the job accordingly. It will spin up one pod to run the job (client/job-runner), one pod for the scheduler and as many worker pods as specified in the Flyte task decorator
+1. While the `dask` task is running, Flyte (propeller) will continously monitor the created `DaskJob` resource, waiting on it to report success or failure. After the job has finished (or the Flyte task has been terminated), all `dask` related resources will get cleaned up
+
+## Useful links
+
+- [Flyte documentation](https://docs.flyte.org/en/latest/)
+- [Flyte community](https://docs.flyte.org/en/latest/community/index.html)
+- [flytekitplugins-dask user documentation](https://docs.flyte.org/projects/cookbook/en/latest/auto/integrations/kubernetes/k8s_dask/index.html)
+- [flytekitplugins-dask deployment documentation](https://docs.flyte.org/en/latest/deployment/plugins/k8s/index.html)
+- [dask-kubernetes documentation](https://kubernetes.dask.org/en/latest/)
+- [Blog post on the dask kubernetes operator](http://blog.dask.org/2022/11/09/dask-kubernetes-operator)
+
+In case there are any questions, don't hesitate to reach out. Either in the [Flyte slack](https://flyte-org.slack.com/join/shared_invite/zt-1joarlg5w-cS7sT~pz0oH6PjyITj3mfg#/shared-invite/email) ("Bernhard Stadlbauer") or on [GitHub](https://github.com/bstadlbauer).
+
+I would like to give a shoutout for all of the help I've received from [Jacob Tomlinson](https://jacobtomlinson.dev/) (Dask) as well as [Dan Rammer](https://github.com/hamersaw) (Flyte). This would not have been possible without your support.
```

Comment:
```suggestion
In case there are any questions or concerns, don't hesitate to reach out. You can contact Bernhard Stadlbauer via the [Flyte Slack](https://slack.flyte.org/) or via [GitHub](https://github.com/bstadlbauer).

I would like to give a shoutout for all of the help I've received from [Jacob Tomlinson](https://jacobtomlinson.dev/) (Dask) as well as [Dan Rammer](https://github.com/hamersaw) (Flyte). This would not have been possible without your support.
```